railties/lib/rails/configuration.rb

```diff
@@ -64,9 +65,10 @@ def unshift(*args, &block)
       end
 
       def merge_into(other) #:nodoc:
-        @operations.each do |operation, args, block|
+        @operations.concat(@delete_operations).each do |operation, args, block|
```

Comment:
Should we be actually changing the operations array? Maybe we should create another array by adding both?